### DISCUSSION TITLE
Building and installing libssh2 on *nix system using Cmake route, makes libssh2 target not globally visible.

### DISCUSSION THREAD

**Discussion by bilal614** [libssh2/libssh2#1150] [Q&A]
I have tried following the INSTALL_CMAKE.md guide for building and installing libssh2 on an Linux based system. The Readme file states that '**_You still have to make libssh2 available on your system first. You can install it in the traditional way shown above..._**' 
'**_cmake <libssh2 source location>
  cmake --build .
  cmake --build . --target install_**'

Then using the prescribed way to find the libssh2 package on the system '**_find_package(libssh2 REQUIRED CONFIG)_**', I run into the following error:
 
![libssh2_issue](https://github.com/libssh2/libssh2/assets/15609233/9dfc9699-3c40-4d7b-95d1-2a6dc6109418)

In the generated libssh2-targets.cmake the libssh2::libssh2_static imported target is not set to GLOBAL. 
![image](https://github.com/libssh2/libssh2/assets/15609233/2928be7b-ffc7-4f9f-8c87-d97aff94ac2c)

Is my usage described above(build and install on system using Cmake as described in the Readme) incorrect? Do I miss something?  

NOTE: 
- Based on static library libssh2 build
- Adding GLOBAL to the generated libssh2-targets.cmake imported target does resolve the issue of finding the libssh2 targets, however not a correct way to go about it

**Comment by vszakats** [comment]
Related to recent 2da13c13e14d990df23686490a1e98d963ea7f71 #1104 where we added `libssh2-targets.cmake`.

Can't say what's broken or how to improve it, but we're certainly open for ideas.

**Comment by vszakats** [comment]
Possibly the same issue [came up](https://github.com/curl/curl/pull/11629#issuecomment-1671596059) with libcurl, with this suggestion: https://github.com/curl/curl/pull/11646

Does this patch make any improvement for you? (beware: untested, feel free to tweak):
```diff
diff --git a/cmake/libssh2-config.cmake.in b/cmake/libssh2-config.cmake.in
index 84b98cb2..45d3919a 100644
--- a/cmake/libssh2-config.cmake.in
+++ b/cmake/libssh2-config.cmake.in
@@ -3,8 +3,5 @@
 
 include("${CMAKE_CURRENT_LIST_DIR}/libssh2-targets.cmake")
 
-# Alias for either shared or static library
-add_library(libssh2::libssh2 ALIAS libssh2::@LIB_SELECTED@)
-
 # Compatibility alias
 add_library(Libssh2::libssh2 ALIAS libssh2::@LIB_SELECTED@)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index cae1f27b..5a954065 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -139,6 +139,9 @@ if(BUILD_SHARED_LIBS)
       $<INSTALL_INTERFACE:$<INSTALL_PREFIX>/${CMAKE_INSTALL_INCLUDEDIR}>)
 endif()
 
+# export libcurl alias as target name
+set_property(TARGET ${LIB_SELECTED} PROPERTY EXPORT_NAME "libssh2")
+
 ## Installation
 
 install(FILES
```

**Reply by vszakats** [reply]
PR: #1154
